src/kdbindings/signal.h

```diff
@@ -242,9 +87,36 @@ class Signal
             return m_connections.insert({ slot });
         }
 
+        // Establish a deferred connection between signal and slot, where ConnectionEvaluator object
+        // is used to queue all the connection to evaluate later. The returned
+        // value can be used to disconnect the slot later.
+        Private::GenerationalIndex connectDeferred(const std::shared_ptr<ConnectionEvaluator> &evaluator, std::function<void(Args...)> const &slot, const ConnectionHandle &handle)
+        {
+            auto weakEvaluator = std::weak_ptr<ConnectionEvaluator>(evaluator);
+
+            auto connection = [weakEvaluator = std::move(weakEvaluator), slot, &handle](Args... args) {
+                // Check if the ConnectionEvaluator is still alive
+                if (auto evaluatorPtr = weakEvaluator.lock()) {
+                    auto lambda = [slot, args...]() {
+                        slot(args...);
+                    };
+                    evaluatorPtr->enqueueSlotInvocation(handle, lambda);
+                } else {
+                    throw std::runtime_error("ConnectionEvaluator is no longer alive");
+                }
+            };
+            return m_connections.insert({ connection, true });
+        }
+
         // Disconnects a previously connected function
-        void disconnect(const Private::GenerationalIndex &id) override
+        void disconnect(const Private::GenerationalIndex &id, const ConnectionHandle &handle) override
         {
+            // If the connection evaluator is still valid, remove any queued up slot invocations
+            // associated with the given handle to prevent them from being evaluated in the future.
+            if (auto evaluatorPtr = m_connectionEvaluator.lock()) {
```

Comment:
This should really check if this specific connection is deferred, so that we can save ourselves a search through the list of queued signal invocations if it's not a deferred signal.

Also see my note on m_connectionEvaluator in general.